risc0/zkvm/src/host/client/env.rs

```diff
@@ -79,6 +79,9 @@ pub struct ProveZkrRequest {
 /// A Keccak proof request.
 #[stability::unstable]
 pub struct ProveKeccakRequest {
+    /// The digest of the claim that this ZKR program is expected to produce.
+    pub claim_digest: Digest,
```

Comment:
Why is this needed for a proof request?

Reply:
originally, I didn't think this was needed either. so I left it out. As i worked on the prover example, I saw that for prove_zkr in the prover example, this was utilized. I was thinking that this could be computed from the receipt produced from the proof but i saw that the prove_zkr did it thsi way so I added this. This is set by the host syscall handler and is passed from the guest. If there's a mismatch then it means that the guest and the prover disagreed on the digest, which is a good check to have. but I guess this spurs another question: why does this exist on the prove_zkr side?